shared/profile/showcase-team-offer/index.js

```diff
@@ -0,0 +1,159 @@
+          type="PrimaryGreen"
+          waiting={waiting}
+        />
+      ) : (
+        <Text style={{color: globalColors.black_40, width: isMobile ? '35%' : '20%'}} type="BodySmall">
+          Admins aren’t allowing members to publish.
+        </Text>
+      )}
+    </Box>
+    {!isMobile && <Divider style={{marginLeft: 48}} />}
+  </Box>
+)
+
+const ShowcaseTeamOffer = (props: Props) => (
+  <Box style={styleContainer}>
+    <Text style={{paddingBottom: globalMargins.small}} type="Header">
+      Publish the teams you're in
+    </Text>
+    <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', marginBottom: globalMargins.xtiny}}>
+      <Box style={{backgroundColor: globalColors.black_05, height: 1, width: 24}} />
+      <Icon
+        style={{
+          color: globalColors.black_10,
+          paddingLeft: globalMargins.tiny,
+          paddingRight: globalMargins.tiny,
+        }}
+        type="iconfont-info"
+      />
+      <Box style={{backgroundColor: globalColors.black_05, height: 1, width: 24}} />
+    </Box>
+
+    <Text
+      style={{color: globalColors.black_40, paddingTop: globalMargins.tiny, textAlign: 'center'}}
+      type="BodySmall"
+    >
+      Promoting a team will encourage others to ask to join.
+    </Text>
+    <Text
+      style={{
+        color: globalColors.black_40,
+        paddingTop: isMobile ? globalMargins.tiny : 0,
+        textAlign: 'center',
+      }}
+      type="BodySmall"
+    >
+      The team's description and number of members will be public.
+    </Text>
+
+    <ScrollView style={{flexShrink: 1, marginTop: globalMargins.medium, width: '100%'}}>
+      {props.teamnames &&
+        props.teamnames.map(name => (
+          <TeamRow
+            canShowcase={
+              props.teamNameToCanPerform[name] && props.teamNameToCanPerform[name].setMemberShowcase
+            }
+            key={name}
+            name={name}
+            isOpen={props.teamNameToIsOpen[name]}
+            membercount={props.teammembercounts[name]}
+            onPromote={promoted => props.onPromote(name, promoted)}
+            showcased={
+              props.teamNameToPublicitySettings[name] && props.teamNameToPublicitySettings[name].member
+            }
+            waiting={!!props.waiting[teamWaitingKey(name)]}
+          />
+        ))}
+    </ScrollView>
+    <ClickableBox onClick={props.onBack} style={{flexGrow: 1}}>
+      <Button style={{margin: globalMargins.small}} type="Secondary" onClick={props.onBack} label="Close" />
+    </ClickableBox>
+  </Box>
+)
+
+const styleContainer = {
+  ...globalStyles.flexBoxColumn,
+  alignItems: 'center',
+  flex: 1,
+  marginTop: 35,
+  marginBottom: isMobile ? globalMargins.xtiny : 55,
+  minWidth: isMobile ? undefined : 600,
+}
+
+const styleMeta = {
+  alignSelf: 'center',
+  backgroundColor: globalColors.green,
+  borderRadius: 1,
+  marginLeft: globalMargins.xtiny,
+  marginTop: 2,
+}
+
+const PopupWrapped = (props: Props) => (
```

Comment:
@akalin another example of crappy stuff we currently do to handle mobile/desktop popup stuff